include/zephyr/kernel.h

```diff
@@ -35,6 +35,11 @@ BUILD_ASSERT(sizeof(int32_t) == sizeof(int));
 BUILD_ASSERT(sizeof(int64_t) == sizeof(long long));
 BUILD_ASSERT(sizeof(intptr_t) == sizeof(long));
 
+#ifdef __cplusplus
```

Comment:
I do not see how this relates to the kernel, can we find different place for this please?

Reply:
@nashif Would you be happy with toolchain.h as @andyross proposed?